include/aws/common/encoding.h

```diff
@@ -0,0 +1,175 @@
+#ifndef AWS_COMMON_ENCODING_H
+#define AWS_COMMON_ENCODING_H
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/common.h>
+#include <stdint.h>
+#include <memory.h>
+
+#ifdef _WIN32
+#include <winsock2.h>
+#else
+#include <netinet/in.h>
+#endif /*_WIN32 */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * computes the length necessary to store the result of aws_hex_encode(). returns -1 on failure, and 0 on success.
+ * encoded_length will be set on success.
+ */
+AWS_COMMON_API int aws_compute_hex_encoded_len(size_t to_encode_len, size_t *encoded_length);
+
+/*
+ * Base 16 (hex) encodes the contents of to_encode and stores the result in output.
+ * If output is NULL, output_size will be set to what the output_size should be.
+ */
+AWS_COMMON_API int aws_hex_encode(const uint8_t *restrict to_encode, size_t to_encode_len, char *restrict output, size_t output_size);
+
+/*
+ * computes the length necessary to store the result of aws_hex_decode(). returns -1 on failure, and 0 on success.
+ * decoded_len will be set on success.
+ */
+AWS_COMMON_API int aws_compute_hex_decoded_len(size_t to_decode_len, size_t *decoded_len);
+
+/*
+ * Base 16 (hex) decodes the contents of to_decode and stores the result in output.
+ * If output is NULL, output_size will be set to what the output_size should be.
+ */
+AWS_COMMON_API int aws_hex_decode(const char *restrict to_decode, size_t to_decode_len, uint8_t *restrict output, size_t output_size);
+
+/*
+ * Computes the length necessary to store the output of aws_base64_encode call. returns -1 on failure, and 0 on success.
+ * encoded_length will be set on success.
+ */
+AWS_COMMON_API int aws_compute_base64_encoded_len(size_t to_encode_len, size_t *encoded_len);
```

Comment:
Would recommend using consistent prefixes - b64 related functions should all start with `aws_base64_`. This makes grepping easier and reduces cognitive burden.